ox_bqpipeline/bqpipeline.py

```diff
@@ -350,72 +341,82 @@ def create_job_config(self, batch=False, dest=None, create=True,
         else:
             data_set = None
 
-        return bigquery.QueryJobConfig(priority=priority,
-                                       default_dataset=data_set,
-                                       destination=dest_tableref,
-                                       create_disposition=create_disp,
-                                       write_disposition=write_disp)
+        job_config_settings = {
+            'priority': priority,
+            'default_dataset': data_set,
+            'destination': dest_tableref,
+            'create_disposition': create_disp,
+            'write_disposition': write_disp,
+        }
+        if query_params:
+            job_config_settings.update({'query_parameters':
+                                        self.set_query_params(query_params)})
 
-    def validate_query_params(self, query_params, depth=0):
+        return bigquery.QueryJobConfig(**job_config_settings)
+
+    def validate_query_params(self, query_params):
         """Validate the named/positional query parameters."""
-        if not (isinstance(query_params, list) or
-                isinstance(query_params, dict)):
+        if not isinstance(query_params, (list, dict)):
             return False
-        for key in query_params:
-            if isinstance(query_params, list):
-                value = key
-                key = None
-            else:
-                if not (depth or isinstance(key, str)):
-                    return False
-                value = query_params[key]
-
+        if isinstance(query_params, list):
+            res = all([self.validate_parameter(value) for value in
+                       query_params])
+        else:
+            datatype = set(map(type, query_params))
+            # Make sure all keys are str
+            res = len(datatype) == 1 and datatype.issubset([str])
+            res = res and all([self.validate_parameter(query_params[key])
+                               for key in query_params])
+        return res
+
+    def validate_parameter(self, query_parameter):
+        """Validate individual query parameter."""
+        res = True
+        if isinstance(query_parameter, list):
             # For list, make sure each element is a scalar type.
+            datatype = set(map(type, query_parameter))
+            res = (len(datatype) == 1 and
+                   datatype.issubset(BQ_SCALAR_TYPE_MAP))
+        elif isinstance(query_parameter, dict):
+            if not query_parameter:
+                return True
+            # For dict, make sure all keys are same type and
+            # one of BQ supported types.
+            datatype = set(map(type, query_parameter))
```

Comment:
This is gonna check the type of the keys in query_parameter. Which is likely just going to be all string.

Reply:
Yes. Fixed.